cosmos/job/drm/drm_k8s_jobs.py

```diff
@@ -0,0 +1,137 @@
+
+        kbatch_cmd = 'kbatch --script {script} {kbatch_option_str}'.format(
+            script=task.output_command_script_path,
+            kbatch_option_str=kbatch_option_str,
+        )
+
+        kbatch_proc = sp.Popen(kbatch_cmd, shell=True, stdout=sp.PIPE, stderr=sp.PIPE)
+        job_id, err = kbatch_proc.communicate()
+
+        if err:
+            raise RuntimeError(err)
+
+        job_id = job_id.decode('utf-8').replace('\n', '')
+
+        task.drm_jobID = job_id
+        task.status = TaskStatus.submitted
+
+    def _get_task_completed_info(self, task, task_infos):
+        task_info = task_infos[task.drm_jobID]
+
+        task_status = task_info['status']
+        if task_status.get('active'):
+            return None
+
+        successful = task_status.get('succeeded')
+        exit_code = 0 if successful else 1
+        start_time = dateutil.parser.parse(task_status['startTime'])
+
+        if successful:
+            end_time_iso8601 = task_status['completionTime']
+        else:
+            failed_info = next(condition for condition in task_status['conditions'] if condition['type'] == 'Failed')
+            end_time_iso8601 = failed_info['lastProbeTime']
+
+        end_time = dateutil.parser.parse(end_time_iso8601)
+
+        wall_time_delta = end_time - start_time
+        wall_time = round(wall_time_delta.total_seconds())
+
+        return dict(exit_status=exit_code, wall_time=wall_time)
+
+    def filter_is_done(self, tasks):
+        task_infos = self.drm_statuses(tasks)
+        for task in tasks:
+            task_completed_info = self._get_task_completed_info(task, task_infos)
+            if task_completed_info:
+                yield task, task_completed_info
+
+    def drm_statuses(self, tasks):
+        job_ids = [task.drm_jobID for task in tasks]
+
+        kstatus_cmd = 'kstatus {job_ids} -o json'.format(job_ids=' '.join(job_ids))
+        kstatus_proc = sp.Popen(kstatus_cmd, shell=True, stdout=sp.PIPE, stderr=sp.PIPE)
+
+        task_infos, err = kstatus_proc.communicate()
+        if err:
+            raise RuntimeError(err)
+
+        task_infos = json.loads(task_infos)
+        if len(job_ids) > 1:
+            task_infos = task_infos['items']
+        else:
+            task_infos = [task_infos]
+
+        task_infos = {task_info['metadata']['labels']['job-name']: task_info for task_info in task_infos}
+        return task_infos
+
+    def kill(self, task):
+        # Transfer the logs from our job into an output file before killing it
+        job_id = task.drm_jobID
+        stream_logs_cmd = 'klogs {job_id}'.format(job_id=job_id)
+
+        sp.Popen(stream_logs_cmd,
+                 stdout=open(task.output_stdout_path, 'w'),
+                 stderr=open(task.output_stderr_path, 'w'),
+                 shell=True)
+
+        kill_cmd = 'kcancel {job_id}'.format(job_id=job_id)
+
+        kill_proc = sp.Popen(kill_cmd, shell=True, stdout=sp.PIPE, stderr=sp.PIPE)
```

Comment:
any reason not to just use check=True?

Reply:
check=True is not available in Python 2.7. check_call and check_output do not return the stderr in the exception, making for a convoluted error message containing simply the exit code. I would have to catch the exception and re-raise it anyway, making the code uglier and more verbose.